lib/autoproj/cli/build.rb

```diff
@@ -42,6 +43,16 @@ def run(selected_packages, options)
                     pkg.disable
                 end
 
+                # and disable packages listed in the not command if there are any
+                if build_options[:not]
+                    source_packages.each do |pkg_name|
```

Comment:
I'd prefer that you integrate this with the loop above, which have the same purpose (disable what's not to be built).

Moreover, don't check for nil in `not`. Make sure that `build_options[:not]` is always set instead.

Reply:
Is there any reason to use `next` in the disabling loop? I was thinking to do something like this:
```
if  source_packages.include?(pkg.name) || 
   (build_options[:not] && !build_options[:not].include?(pkg.name))
            Autobuild.warn 'disabling ' + pkg.name
            pkg.disable
 end
```

Reply:
I have refactored the loops with a single one using an active_packages variable that contains source packages minus packages in the not option if there are any.